### PR TITLE
Fix Snapshot Status API Docs Test

### LINKED ISSUE #59767: [CI] Failures in DocsClientYamlTestSuiteIT {yaml=reference/snapshot-restore/apis/get-snapshot-status-api/line_309}
**Build scan**:

https://gradle-enterprise.elastic.co/s/gs2c3e636a73s
https://gradle-enterprise.elastic.co/s/di5pmdh26jj2y

**Repro line**:

    ./gradlew ':docs:integTestRunner' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/snapshot-restore/apis/get-snapshot-status-api/line_309}" -Dtests.seed=580EE8BCEA3F32A5 -Dtests.security.manager=true -Dtests.locale=ar-BH -Dtests.timezone=Etc/GMT+1 -Druntime.java=11

**Reproduces locally?**:

Yes!

**Applicable branches**:

`master`

**Failure history**:

[Gradle enterprise history](https://gradle-enterprise.elastic.co/scans/tests?search.buildToolType=gradle&search.buildToolType=maven&search.relativeStartTime=P7D&search.timeZoneId=Europe/London&tests.container=org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT&tests.sortField=FAILED&tests.test=test%20%7Byaml%3Dreference/snapshot-restore/apis/get-snapshot-status-api/line_309%7D&tests.unstableOnly=true)

**Failure excerpt**:

<details>
<summary>Stack trace / error</summary>

```
        java.lang.AssertionError: $body didn't match expected value:
                                 $body: 
                               snapshots: 
                                         0: 
                        include_global_state: same [false]
                                     indices: 
                                       index_1: 
                                          shards: 
                                                 0: 
                                               stage: same [DONE]
                                               stats: 
                                           incremental: 
                                              file_count: same [0]
                                           size_in_bytes: same [0]
                                  start_time_in_millis: same [1594989648341]
                                        time_in_millis: same [0]
                                                 total: 
                                              file_count: expected Integer [4] but was Integer [7]
                                           size_in_bytes: same [11590]
                                    shards_stats: 
                                              done: same [1]
                                            failed: same [0]
                                        finalizing: same [0]
                                      initializing: same [0]
                                           started: same [0]
                                             total: same [1]
                                           stats: 
                                       incremental: 
                                          file_count: same [0]
                                       size_in_bytes: same [0]
                              start_time_in_millis: same [1594989648341]
                                    time_in_millis: same [0]
                                             total: 
                                          file_count: expected Integer [4] but was Integer [7]
                                       size_in_bytes: same [11590]
                                  repository: same [my_repository]
                                shards_stats: 
                                          done: same [1]
                                        failed: same [0]
                                    finalizing: same [0]
                                  initializing: same [0]
                                       started: same [0]
                                         total: same [1]
                                    snapshot: same [snapshot_2]
                                       state: same [SUCCESS]
                                       stats: 
                                   incremental: 
                                      file_count: same [0]
                                   size_in_bytes: same [0]
                          start_time_in_millis: same [1594989648341]
                                time_in_millis: same [201]
                                         total: 
                                      file_count: expected Integer [4] but was Integer [7]
                                   size_in_bytes: same [11590]
                                        uuid: same [gHeds_gQSrubBFRlf7MUvg]
```

</details>

### PR DESCRIPTION
**Pull request by original-brownbear**
We can't just assume a fixed number for the overall file count.
Depending on how the merging/flushing works out we won't always have
4 files for the index across all versions, systems etc.
Also, we could have x-pack concurrently create some system indices
which could mess up the total numbers here.
Fixed by only snapshotting a single index+shard in the snapshot that
we get the status for and verifying consistency instead of equality
for total file counts.

Closes #59767

Backports:
* 7.x #59787
* 7.9 #59789
* 7.8 #59790

### CODE REVIEW

**Review by jrodewig — APPROVED** [review]
👍  LGTM.

Thanks for fixing. I hadn't considered that this stat may not be predictable.

**Review by lockewritesdocs — APPROVED** [review]
@original-brownbear, thank you for fixing this issue! It took some iterating with @jrodewig to fix the original test cases. The tests passed the CI checks, so I'm curious why it's now causing errors. I'll backport this change after it's merged.

**Comment by original-brownbear** [comment]
Thanks @jrodewig @lockewritesdocs !


>  I'll backport this change after it's merged.

Thanks, then I won't backport here and leave it to you :)

**Pull request merged**
